tensorboard/webapp/metrics/types.ts

```diff
@@ -46,8 +46,33 @@ export type NonPinnedCardId = string;
 
 export type PinnedCardId = string;
 
+/**
+ * A unique identifier to a specific card instance in the UI. This is an opaque
+ * ID, meaning that consumers should never peer into/parse it and never assume
+ * that it will always be a string.
+ */
 export type CardId = NonPinnedCardId | PinnedCardId;
 
 export type CardIdWithMetadata = CardMetadata & {
   cardId: CardId;
 };
+
+/**
+ * The most minimal representation of a card that uniquely identifies it.
+ * This information can be used in storage layers (e.g. URL) and may be matched
```

Comment:
> This information may be persisted...

Done

> Also, maybe it is worth noting potential staleness/incorrectness (since user may edit it by hand) of the data

Same question, are we referring to the case of deleting runs only?